test/Microsoft.DotNet.Tools.Publish.Tests/Microsoft.DotNet.Tools.Publish.Tests.cs

```diff
@@ -148,6 +145,34 @@ public void CompilationFailedTest()
             publishCommand.Execute().Should().Fail();
         }
 
+        [Fact]
+        public void PublishScriptsRun()
+        {
+            // create unique directories in the 'temp' folder
+            var root = Temp.CreateDirectory();
+            var testAppDir = root.CreateDirectory("TestApp");
+            var testLibDir = root.CreateDirectory("TestLibrary");
+
+            //copy projects to the temp dir
+            CopyProjectToTempDir(Path.Combine(_testProjectsRoot, "TestApp"), testAppDir);
+            CopyProjectToTempDir(Path.Combine(_testProjectsRoot, "TestLibrary"), testLibDir);
+
+            RunRestore(testAppDir.Path);
+            RunRestore(testLibDir.Path);
+
+            // run publish
+            var testProject = GetProjectPath(testAppDir);
+            var publishCommand = new PublishCommand(testProject);
+
+            var result = publishCommand.ExecuteWithCapturedOutput();
+
+            Assert.True(
+                Regex.Match(result.StdOut, "\nprepublish_output( '[^%]+'){5}.+\npostpublish_output( '[^%]+'){5}",
+                    RegexOptions.Singleline).Success);
```

Comment:
We are trying to use FluentAssertions style assertions in the tests. You can see an example for asserting stdout [here](https://github.com/dotnet/cli/blob/02d28537f89ddb3908f6f2db90d68da36f36751e/test/E2E/EndToEndTest.cs#L144).

Can you add support for regex matching in the [assertions file](https://github.com/dotnet/cli/blob/02d28537f89ddb3908f6f2db90d68da36f36751e/test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/CommandResultAssertions.cs)?